stan/math/fwd/fun/mdivide_left.hpp

```diff
@@ -31,17 +39,19 @@ inline Eigen::Matrix<fvar<T>, R1, C2> mdivide_left(
   Eigen::Matrix<T, R2, C2> val_b(b.rows(), b.cols());
   Eigen::Matrix<T, R2, C2> deriv_b(b.rows(), b.cols());
 
+  const Eigen::Ref<const plain_type_t<T1>>& A_ref = A;
   for (int j = 0; j < A.cols(); j++) {
     for (int i = 0; i < A.rows(); i++) {
-      val_A(i, j) = A(i, j).val_;
-      deriv_A(i, j) = A(i, j).d_;
+      val_A(i, j) = A_ref(i, j).val_;
```

Comment:
I am pretty sure if a do `val_A = A.val(); deriv_A = A.d()`, A would be evaluated twice. I think we can not avoid either a copy or two evaluations.

Reply:
Ah gotcha, makes sense

Reply:
Hey, I think you have some experience with adding to Eigen library. Do you think it would be possible to make some kind of expression that would make a `fvar` "Matrix" by referencing two matrices of `double`s. So that we could directley evaluate a fvar expression into two matrices like this:
```
make_fvar_ref_matrix(val_A, deriv_A) = A.eval();
```

Reply:
Yep I could do that. I think the easiest approach would be a custom ```CwiseBinaryOp```, something like:
```
val_A.make_fvar(deriv_A)
```
I could just drop that into the current matrixbase addons. The tricky part will be the order of includes, since the ```fvar``` header will need to be included before the ```eigen_plugins.h``` header. but I can do some testing and see what works.